TESTING.md

```diff
@@ -26,82 +26,144 @@ Unit tests are used to test particular code snippets in isolation. They are your
 
 ```shell
 podman machine init --cpus=4 --memory=8192 --rootful --now
 ```
 
-Before running automated end-to-end tests, you need run the following command to make images and load it in your local cluster:
+Before running automated end-to-end tests, you need to run the following command to make images and load it in your local cluster:
 
 ```shell
 CONTAINER_CLI=podman KIND_CONTEXT=e2e make test-e2e-images
 ```
 
+> [!TIP]
+> If you are using [podman-desktop](https://podman-desktop.io/docs/podman/creating-a-podman-machine), it is quite easy to create the podman machine or update the podman machine resource settings. It also has a nice [kind extension](https://podman-desktop.io/docs/kind) to make the intergation with kind easier.
+
+#### Minikube
+
+Create the minikube cluster.
+
+```shell
+minikube start --embed-certs --profile e2e
+```
+
+Above command will use the default driver. If you want to use a different driver, use `--driver` flag
+in the command. Refer [minikube documentation](https://minikube.sigs.k8s.io/docs/drivers/) for more details.
+
+> [!NOTE]
+> `--embed-certs` option is needed to embed the certs in kubeconfig like in Kind and extract it later for
+> running the script `./scripts/run-external.sh`
+
+##### Building images and loading them into your minikube cluster
+
+###### Using docker with minikube
+
+Before running automated end-to-end tests, you need run the following command to make images and load it in your local cluster:
+
+```shell
+make test-e2e-images-minikube
+```
+
+###### Using podman with minikube
+
+Follow the same instructions from [using podman with kind](#using-podman-with-kind) for creating podman machine.
+
+Before running automated end-to-end tests, you need to run the following command to make images and load it in your local cluster:
+
+```shell
+CONTAINER_CLI=podman make test-e2e-images-minikube
+```
+
+> [!TIP]
+> If you are using [podman-desktop](https://podman-desktop.io/docs/podman/creating-a-podman-machine), it is quite easy to create the podman machine or update the podman machine resource settings. It also has a nice [minikube extension](https://podman-desktop.io/docs/minikube) to make the intergation with minikube easier.
+
+### Troubleshooting
+
+If you get errors while running the `make` commands, most probably running `make clean` and then executing commands will fix
+the issues.
+
+#### Manual Testing
+
+For manual testing, you can use the utility script [scripts/run-external.sh](scripts/run-external.sh), it will check all the requirements and run your local version of the Prometheus Operator on your cluster:
+
+If the default context already points to the test cluster created, you can simply run:
+
+```shell
+./scripts/run-external.sh -c
+```
+
+else run:
+
+```shell
+./scripts/run-external.sh <cluster context name>
+```
+
 ### Running the automated E2E Tests
```

Comment:
(nit) should we reverse the order and go from running one specific e2e test to running a subset and then running everything? The rationale being that more than often, the CI will fail on 1 specific e2e test and you want to debug locally.